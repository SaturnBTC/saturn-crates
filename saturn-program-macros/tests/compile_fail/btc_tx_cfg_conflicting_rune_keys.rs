use saturn_program_macros::saturn_program;

mod instruction {
    #[derive(borsh::BorshSerialize, borsh::BorshDeserialize)]
    pub enum Instr {
        Dummy,
    }
}

#[saturn_program(
    btc_tx_cfg(max_inputs_to_sign = 1, max_modified_accounts = 1, rune_set = "crate::Foo", rune_capacity = 3)
)]
mod handlers {}

fn main() {} 