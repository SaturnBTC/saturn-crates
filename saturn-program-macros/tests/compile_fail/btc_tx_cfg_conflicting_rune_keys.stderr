error: btc_tx_cfg keys rune_set and rune_capacity are mutually exclusive
  --> tests/compile_fail/btc_tx_cfg_conflicting_rune_keys.rs:10:1
   |
10 | / #[saturn_program(
11 | |     btc_tx_cfg(max_inputs_to_sign = 1, max_modified_accounts = 1, rune_set = "crate::Foo", rune_capacity = 8)
12 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `saturn_program` (in Nightly builds, run with -Z macro-backtrace for more info)
